src/textual/dom.py

```diff
@@ -334,7 +335,10 @@ def setter(value: object) -> None:
                     """Set bound data."""
                     _rich_traceback_omit = True
                     Reactive._initialize_object(self)
-                    setattr(self, variable_name, value)
+                    # Copy the bound value
```

Comment:
I feel like this should be documented since copying can be expensive or potentially crash the program.

I'm guessing shallow copying means it also wouldn't fix the confusion for e.g. a list of mutable objects. For example, if one of the objects in the list is mutated and I call `mutate_reactive`, the prior confusing scenario still arises.

Reply:
I'm reconsidering the copy.

It's not clear what the dev would expect when modifying a child attribute that has been bound. I'm not sure if they would expect the changes to be reflected on the parent. But I don't know if the would expect the inverse either.

Going to remove the copy, until I have a better idea what it would fix (if anything)